tests/testthat/test-cdisc_data.R

```diff
@@ -0,0 +1,267 @@
+  testthat::expect_silent(
+    cdisc_data(
+      cdisc_dataset("ADSL", adsl_raw),
+      cdisc_dataset("ARG1", adsl_raw, keys = get_cdisc_keys("ADSL")),
+      cdisc_dataset("ARG2", adsl_raw, keys = get_cdisc_keys("ADSL")),
+      code = "ADSL <- ARG1 <- ARG2 <- as.data.frame(as.list(setNames(nm = get_cdisc_keys(\"ADSL\"))))",
+      check = TRUE
+    )
+  )
+
+  testthat::expect_error(
+    cdisc_data(
+      cdisc_dataset(
+        "ADSL",
+        adsl_raw,
+        code = "ADSL <- as.data.frame(as.list(setNames(nm = get_cdisc_keys(\"ADSL\"))))"
+      ),
+      cdisc_dataset("ARG1", adsl_raw, keys = get_cdisc_keys("ADSL")),
+      cdisc_dataset("ARG2", adsl_raw, keys = get_cdisc_keys("ADSL")),
+      code = "test",
+      check = TRUE
+    ),
+    "'code' argument should be specified only in the 'cdisc_data' or in 'cdisc_dataset' but not in both"
+  )
+
+  testthat::expect_error(
+    cdisc_data(
+      cdisc_dataset(
+        "ADSL",
+        adsl_raw,
+        code = "ADSL <- as.data.frame(as.list(setNames(nm = get_cdisc_keys(\"ADSL\"))))"
+      ),
+      cdisc_dataset(
+        dataname = "ARG1",
+        x = dplyr::mutate(adsl_raw, x1 = 1),
+        keys = get_cdisc_keys("ADSL"),
+        code = "ARG1 <- as.data.frame(as.list(setNames(nm = get_cdisc_keys(\"ADSL\"))))"
+      ),
+      cdisc_dataset(
+        "ARG2",
+        adsl_raw,
+        keys = get_cdisc_keys("ADSL"),
+        code = "ARG2 <- as.data.frame(as.list(setNames(nm = get_cdisc_keys(\"ADSL\"))))"
+      ),
+      check = TRUE
+    ),
+    "Reproducibility check failed."
+  )
+})
+
+testthat::test_that("List values", {
+  test_relational_data_equal <- function(data1, data2) {
+    testthat::expect_equal(data1$get_items(), data2$get_items())
+    testthat::expect_equal(data1$get_join_keys(), data2$get_join_keys())
+    mapply(testthat::expect_equal, data1$get_ui("test"), data2$get_ui("test"))
+  }
+
+  result <- cdisc_data(cdisc_dataset("ADSL", adsl_raw, label = "test_label"))
+
+  datasets <- list(cdisc_dataset(
+    dataname = "ADSL",
+    x = adsl_raw,
+    keys = c("STUDYID", "USUBJID"),
+    parent = character(0),
+    label = "test_label"
+  ))
+
+  result_to_compare <- do.call(cdisc_data, datasets)
+
+  test_relational_data_equal(result, result_to_compare)
+
+  result <- cdisc_data(cdisc_dataset("ADSL", adsl_raw), cdisc_dataset("ADTTE", adtte_raw))
+
+  datasets <- list(
+    cdisc_dataset(
+      dataname = "ADSL",
+      x = adsl_raw,
+      keys = c("STUDYID", "USUBJID"),
+      parent = character(0),
+      label = character(0)
+    ),
+    cdisc_dataset(
+      dataname = "ADTTE",
+      x = adtte_raw,
+      keys = c("STUDYID", "USUBJID", "PARAMCD"),
+      parent = "ADSL",
+      label = character(0)
+    )
+  )
+
+  result_to_compare <- do.call("cdisc_data", datasets)
+
+  test_relational_data_equal(result, result_to_compare)
+})
+
+testthat::test_that("cdisc_data_file loads the TealData object", {
+  tmp_file <- tempfile(fileext = ".R")
```

Comment:
cdisc_data file tests need the lifecycle verbosity I think https://lifecycle.r-lib.org/reference/verbosity.html

Reply:
added.